arangod/Pregel/ResultActor.h

```diff
@@ -0,0 +1,118 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Volmer
+////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "Actor/ActorPID.h"
+#include "Actor/HandlerBase.h"
+#include "Pregel/ResultMessages.h"
+#include "fmt/core.h"
+
+namespace arangodb::pregel {
+
+struct ResultState {
+  ResultT<PregelResults> results = {TRI_ERROR_NO_ERROR};
+  bool finished{false};
+};
+
+template<typename Inspector>
+auto inspect(Inspector& f, ResultState& x) {
+  return f.object(x).fields(f.field("results", x.results));
+}
+
+template<typename Runtime>
+struct ResultHandler : actor::HandlerBase<Runtime, ResultState> {
+  auto operator()(message::ResultStart start) -> std::unique_ptr<ResultState> {
+    LOG_TOPIC("4a414", INFO, Logger::PREGEL)
+        << fmt::format("Result Actor {} started", this->self);
+    return std::move(this->state);
+  }
+
+  auto operator()(message::SaveResults start) -> std::unique_ptr<ResultState> {
+    this->state->results = {start.results};
+    this->state->finished = true;
+    return std::move(this->state);
+  }
+
+  auto operator()(message::AddResults start) -> std::unique_ptr<ResultState> {
```

Comment:
I just noticed that we are missing a check at the start of this function:
```
if (this->state->finished) {
  return std::move(this->state);
}
```
This is needed for the single server case: There, worker and conductor actors are on the same server and we only have one Result Actor. Now we get two messages in this actor: One SaveResults message and one AddResults message. With this check we make sure that the final results are just set when this actor gets the SaveResults message.